### PR TITLE
SOLR-14759 fix tests that need on lucene test-src

### PR DESCRIPTION
**Pull request by madrob**
Rewrite one, ignore the other two.

### CODE REVIEW

**Review by dweiss — APPROVED** [review]

File: solr/core/src/test/org/apache/solr/search/TestXmlQParser.java [hunk 1]
```diff
@@ -18,20 +18,23 @@
...
 
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
+import org.junit.Ignore;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-
-public class TestXmlQParser extends TestCoreParser {
+@Ignore("Was relying on Lucene test sources. Should copy?")
```
Comment (cpoerschke) [inline solr/core/src/test/org/apache/solr/search/TestXmlQParser.java:36]: SolrCoreParser extends (Lucene) CoreParser. (Solr) XmlQParser[Plugin] uses SolrCoreParser. The extending and use are (currently only) to support user-customisation via configuration of the classes, hence the `testSomeOtherQuery()` below is commented out and the `TestXmlQParser` test here doesn't test anything beyond what it inherits from the `TestCoreParser` test.

I think it's okay to just delete the `TestXmlQParser` test.

code links:

* https://github.com/apache/lucene-solr/blob/releases/lucene-solr%2F8.8.1/lucene/queryparser/src/java/org/apache/lucene/queryparser/xml/CoreParser.java
* https://github.com/apache/lucene-solr/blob/releases/lucene-solr%2F8.8.1/lucene/queryparser/src/test/org/apache/lucene/queryparser/xml/TestCoreParser.java

* https://github.com/apache/lucene-solr/blob/releases/lucene-solr%2F8.8.1/solr/core/src/java/org/apache/solr/search/SolrCoreParser.java
* https://github.com/apache/lucene-solr/blob/releases/lucene-solr%2F8.8.1/solr/core/src/test/org/apache/solr/search/TestSolrCoreParser.java

* https://github.com/apache/lucene-solr/blob/releases/lucene-solr%2F8.8.1/solr/core/src/java/org/apache/solr/search/XmlQParserPlugin.java
* https://github.com/apache/lucene-solr/blob/releases/lucene-solr%2F8.8.1/solr/core/src/test/org/apache/solr/search/TestXmlQParserPlugin.java

Comment (dweiss) [reply solr/core/src/test/org/apache/solr/search/TestXmlQParser.java:36]: Thanks Christine. I'll just remove it.

Comment (cpoerschke) [reply solr/core/src/test/org/apache/solr/search/TestXmlQParser.java:36]: Just opened #2465 for the removal before seeing your comment here (but the JIRA ticket).

**Comment by dweiss** [comment]
I'll remove it in this PR:
https://github.com/apache/lucene-solr/pull/2448

**Pull request merged**